### ISSUE TITLE
Have you considered a monorepo?

### ISSUE THREAD

**Issue by bebraw** [evilsoft/crocks#116]
Hi,

Just came by your library through a chance. The functionality looks great. I started to wonder, though, if it would make sense to port it to monorepo style. You would end up with `@crocks/combinators` and so on. Note that npm made organizations free for open source projects a while ago and you can find tools like [Lerna](https://lernajs.io/) for managing the scheme.

The added benefit would be that this would allow you to split up the current README into something more focused per package. The main README of the project would give an overview of what the library is about and give your documentation some abstraction. You can also aggregate all this as a neat documentation site quite easily.

I wrote [a little blog post](https://medium.com/@bebraw/the-case-for-monorepos-907c1361708a) if you want to learn more about the approach.

Keep up good work. 👍 

PS. Let me know if you want to do a little interview for the SurviveJS blog. I've been highlighting gems like this there every once in a while. 😄

**Comment by evilsoft** [comment]
Thanks so much for your random stumble.

I am doing more than considering, it is on the roadmap. I plan to get it out after a few major versions if all goes well after 1.x.x.

I expect much churn on the initial API and inner-dependencies (?) and want to get that solid in one shot instead of managing those change across multiple projects. So while it will be a bit bloated until around 4.x (I hope) it will allow us to get there quicker IMO.

I do not mind doing an interview about `crocks` or whateves, but I am a pretty boring bloke, so I do not know if your readers would end up hating you for wasting your time on me. But I am game for just about anything.

**Comment by bebraw** [comment]
> I expect much churn on the initial API and inner-dependencies (?) and want to get that solid in one shot instead of managing those change across multiple projects. So while it will be a bit bloated until around 4.x (I hope) it will allow us to get there quicker IMO.

Yeah, I went from monolith to monorepo in Reactabular. Even though there was churn, I think the move was worth it as it forced me to re-evaluate the design. As a result I ended up with something decoupled. The interesting thing to notice was that the move allowed me to push secondary functionality that wasn't related to the core outside of the monorepo so I ended up with a mixed model.

> I do not mind doing an interview about crocks or whateves, but I am a pretty boring bloke, so I do not know if your readers would end up hating you for wasting your time on me. But I am game for just about anything.

Ok, cool. I sent you mail about this.

Feel free to close the issue unless you want to keep this open till monorepo is done. 👍

**Comment by evilsoft** [comment]
Just stated the first steps into mono repoing this silly project.
It all starts with [this PR](https://github.com/evilsoft/crocks/pull/132).

I did not forget about the oppertunity to do an interview on this lib @bebraw, just want to get some things solid before I start drawing attention to this project. Soon, I promise if the offer still stands.

**Comment by bebraw** [comment]
> I did not forget about the oppertunity to do an interview on this lib @bebraw, just want to get some things solid before I start drawing attention to this project. Soon, I promise if the offer still stands.

Sure. Take your time.
